Userland/Libraries/LibC/stdio.cpp

```diff
@@ -424,9 +425,10 @@ void FILE::Buffer::did_dequeue(size_t actual_size)
 {
     VERIFY(actual_size > 0);
 
-    if (m_ungotten) {
-        VERIFY(actual_size == 1);
-        m_ungotten = false;
+    if (m_ungotten != 0u) {
+        VERIFY(actual_size <= static_cast<size_t>(__builtin_popcount(m_ungotten & ungotten_mask)));
```

Comment:
This also has a generic version:

```suggestion
        VERIFY(actual_size <= static_cast<size_t>(popcount(m_ungotten & ungotten_mask)));
```

Reply:
Doesn't look like it.
```
/home/Test/Workspace/serenity/Userland/Libraries/LibC/stdio.cpp:429:51: error: 'popcount' was not declared in this scope
  429 |         VERIFY(actual_size <= static_cast<size_t>(popcount(m_ungotten & ungotten_mask)));
```